plugins/s3_auth/s3_auth.cc

```diff
@@ -1053,19 +1053,18 @@ int
 config_reloader(TSCont cont, TSEvent event, void *edata)
 {
   Dbg(dbg_ctl, "reloading configs");
-  S3Config *s3          = static_cast<S3Config *>(TSContDataGet(cont));
+  S3Config *s3 = static_cast<S3Config *>(TSContDataGet(cont));
+  std::unique_lock lock(s3->reload_mutex);
+  s3->check_current_action(edata);
```

Comment:
Why did you move this call to be first?

Reply:
This is the main change. Prior to this change, the handling action is not cleared if the config is invalid ( returning without clearing the `_conf_rld_action` in line 1061 )